src/domain/usecases/user-monitoring/permission-fixer/RunUserPermissionUseCase.ts

```diff
@@ -322,6 +332,24 @@ export class RunUserPermissionUseCase {
         return processedUsers;
     }
 
+    private addMinimalUserGroupToUsersWithoutUserGroup(
+        allUsers: PermissionFixerUser[],
+        completeTemplateGroups: PermissionFixerTemplateGroupExtended[],
+        minimalGroupId: string
+    ) {
+        allUsers.map(user => {
+            const templateGroupMatch = completeTemplateGroups.find(template => {
+                return user.userGroups.some(
+                    userGroup => userGroup != undefined && template.group.id == userGroup.id
+                );
+            });
+            if (templateGroupMatch == undefined) {
+                //template not found -> all roles are invalid except the minimal role
+                user.userGroups.push({ id: minimalGroupId, name: "Minimal Group" });
+            }
+        });
+    }
+
```

Comment:
A return-less map is a `forEach`, in this case its better to use a functional approach, that way its clear that we are modifying allUsers. 
```typescript
private addMinimalUserGroupToUsersWithoutUserGroup(
    allUsers: PermissionFixerUser[],
    completeTemplateGroups: PermissionFixerTemplateGroupExtended[],
    minimalGroupId: string
) {
    return allUsers.map(user => {
        const templateGroupMatch = completeTemplateGroups.find(template => {
            return user.userGroups.some(
                userGroup => userGroup != undefined && template.group.id == userGroup.id
            );
        });
        if (templateGroupMatch == undefined) {
            //template not found -> all roles are invalid except the minimal role
            user.userGroups.push({ id: minimalGroupId, name: "Minimal Group" });
        }
        // this performs a **shallow** copy, in this case it should be ok. But sometimes you need a deep copy.
        return {...user};
    });
}
``` 

In case of `validateUsers` function, there is no return, so a forEach is in order.
The templateGroupMatch find can be a private method or a function to avoid repeating code.